csharp/autobuilder/Semmle.Autobuild.Shared/BuildCommandAutoRule.cs

```diff
@@ -7,11 +7,11 @@ namespace Semmle.Autobuild.Shared
     /// <summary>
     /// Auto-detection of build scripts.
     /// </summary>
-    public class BuildCommandAutoRule : IBuildRule
+    public class BuildCommandAutoRule : IBuildRule<AutobuildOptionsShared>
```

Comment:
I'm not really familiar with C#, but in java we'd do something like this since I think we want to handle subclasses of `AutobuildOptionsShared`:

```suggestion
    public class BuildCommandAutoRule<T> : IBuildRule<T extends AutobuildOptionsShared>
```

This is more a question than a suggestion.

Reply:
In this case `BuildCommandAutoRule` only needs access to the common options, so there is no need to make it more general.